feat(cli): prompt user for authentication on subsequent logins (#11004, #9329)

elasticspoon opened the pull request:
Fixes #11004 (partially)

### `coder login` When Already Authenticated

 Currently if a user runs `coder login` when logged in they will get the default login flow:

```console
$ > coder login http://localhost:8080
Attempting to authenticate with argument URL: 'http://localhost:8080'
Your browser has been opened to visit:

        http://localhost:8080/cli-auth

> Paste your token here: DqSDeIXhgx-ABpLuIJ5PnLJjtvdUwmRK5
> Welcome to Coder, admin! You're authenticated.
```

Instead prompt the user if they want to re-authenticate if they run `coder login` when already authenticated. Authenticated is defined as having a valid session token stored in the config directory.

```console
$ > coder login http://localhost:8080
Attempting to authenticate with argument URL: 'http://localhost:8080'
> You are already authenticated admin. Are you sure you want to log in again? (yes/no) no
```

Does not apply if they provided a **valid** token via flag or ENV variable. A new token should be generated an stored in that situation unless the `--use-token-as-session` flag was also used.

### Invalid Token

~If an invalid token was set as `ENV` variable a user would be unable to login until the `ENV` variable was cleared. (The same behavior will occur for invalid flags)~

~User will now be informed that the token is invalid and prompted to login normally. The ENV variable won't be cleared but a valid session token will be created.~

I was incorrect about this part. While we can create a new **valid** token at login subsequent commands will still prefer the **invalid** ENV token. A potential solution would be to check if the ENV token if valid and fall back to the config token in that situation, however, that creates the overhead of an additional API call on every authed CLI command invocation.

elasticspoon committed:
feat(cli): prompt when relogging as authed user (#11004, #9329)

Prompt the user if they want to re-authenticate if they run
`coder login` when already authenticated.

Does not apply if they provided a token via flag or ENV
variable. A new token should be generated an stored in that situation
unless the `--use-token-as-session` flag was also used.

fix(cli): prompt user on login attempt with invalid ENV token

Fixes issue where if invalid token was set as ENV
variable a user would be unable to login until the ENV
variable was cleared.

User will now be informed that the token is invalid and
prompted to login normally.

mafredri commented on cli/login.go:
```diff
@@ -274,7 +274,59 @@ func (r *RootCmd) login() *serpent.Command {
 				return nil
 			}
 
+			var userResp codersdk.User
```
Nit: This doesn't seem to be used until later, so we could declare it further down.

mafredri reviewed (commented):
Just a few minor nits but LGTM otherwise, thanks for this PR!

elasticspoon committed:
fixup!: handle cliui.Canceled error

cliui should return nil when returning cliui.Canceled (CTRL-C) and
return error others

fix nits

elasticspoon committed:
fixup!: remove check for ENV token validity in login

I made an incorrect assumption, we can check if
a flag/env token is valid on login and create a new
token if the provided one is invalid.

However, on subsequent requests the invalid ENV token
will still take priority over the valid token stored
on disk. Therefore, this behavior does not work.
